alfresco-health-processor-platform/src/test/java/eu/xenit/alfresco/healthprocessor/plugins/ContentValidationHealthProcessorPluginTest.java

```diff
@@ -0,0 +1,262 @@
+
+        when(nodeService.exists(NODE_REF)).thenReturn(false);
+        assertThat(plugin.process(NODE_REF), is(nullValue()));
+
+        when(nodeService.exists(NODE_REF)).thenReturn(true);
+        when(nodeService.getNodeStatus(NODE_REF)).thenReturn(new Status(100L, NODE_REF, null, 10L, true));
+        assertThat(plugin.process(NODE_REF), is(nullValue()));
+    }
+
+    @Test
+    void process_propertyHasNoContentReader() {
+        ContentValidationHealthProcessorPlugin plugin = initialize(Collections.singletonList(Q_NAME));
+
+        when(nodeService.exists(NODE_REF)).thenReturn(true);
+        when(nodeService.getNodeStatus(NODE_REF)).thenReturn(new Status(100L, NODE_REF, null, 10L, false));
+        when(contentService.getReader(NODE_REF, Q_NAME)).thenReturn(null);
+
+        NodeHealthReport report = plugin.process(NODE_REF);
+        assertThat(report, is(notNullValue()));
+        assertThat(report.getStatus(), is(equalTo(NodeHealthStatus.NONE)));
+        assertThat(report.getNodeRef(), is(equalTo(NODE_REF)));
+    }
+
+    @Test
+    void process_propertyHasContentReader_contentExists() {
+        ContentValidationHealthProcessorPlugin plugin = initialize(Collections.singletonList(Q_NAME));
+
+        when(nodeService.exists(NODE_REF)).thenReturn(true);
+        when(nodeService.getNodeStatus(NODE_REF)).thenReturn(new Status(100L, NODE_REF, null, 10L, false));
+        when(contentService.getReader(NODE_REF, Q_NAME)).thenReturn(new MockContentReader(true));
+
+        NodeHealthReport report = plugin.process(NODE_REF);
+        assertThat(report, is(notNullValue()));
+        assertThat(report.getStatus(), is(equalTo(NodeHealthStatus.HEALTHY)));
+        assertThat(report.getNodeRef(), is(equalTo(NODE_REF)));
+    }
+
+    @Test
+    void process_propertyHasContentReader_contentDoesNotExist() {
+        ContentValidationHealthProcessorPlugin plugin = initialize(Collections.singletonList(Q_NAME));
+
+        when(nodeService.exists(NODE_REF)).thenReturn(true);
+        when(nodeService.getNodeStatus(NODE_REF)).thenReturn(new Status(100L, NODE_REF, null, 10L, false));
+        when(contentService.getReader(NODE_REF, Q_NAME)).thenReturn(new MockContentReader(false));
+
+        NodeHealthReport report = plugin.process(NODE_REF);
+        assertThat(report, is(notNullValue()));
+        assertThat(report.getStatus(), is(equalTo(NodeHealthStatus.UNHEALTHY)));
+        assertThat(report.getNodeRef(), is(equalTo(NODE_REF)));
+        assertThat(report.getMessages(), contains(Q_NAME.toString()));
+    }
+
+    private ContentValidationHealthProcessorPlugin initialize(Collection<QName> propertyQNamesToValidate) {
+        return new ContentValidationHealthProcessorPlugin(nodeService, contentService, dictionaryService,
+                propertyQNamesToValidate);
+    }
+
+    @AllArgsConstructor
+    private static class MockContentReader implements ContentReader {
```

Comment:
That is a lot of boilerplate.

Would having something like this suffice?
```
ContentReader contentReader = mock(ContentReader.class);  
when(contentReader.exists()).thenReturn(false);
```

Reply:
Great remark. 

I always thought it was not possible to make a mock return another mock. However we already do that in these tests so it's possible :man_shrugging: .